src/overlays/actors/ovl_En_Xc/z_en_xc.c

```diff
@@ -77,479 +47,2365 @@ static ColliderCylinderInitType1 sCylinderInit = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Xc/func_80B41844.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Xc/EnXc_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Xc/EnXc_Init.s")
+static u64* sEyeTextures[] = {
+    gShiekEyeOpenTex,
+    gShiekEyeHalfClosedTex,
+    gShiekEyeShutTex,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Xc/func_80B41A88.s")
+void EnXc_InitCollider(Actor* thisx, GlobalContext* globalCtx) {
+    EnXc* this = THIS;
+
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinderType1(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+}
+
+void EnXc_UpdateCollider(Actor* thisx, GlobalContext* globalCtx) {
+    EnXc* this = THIS;
+    Collider* colliderBase = &this->collider.base;
+    s32 pad[3];
+
+    Collider_UpdateCylinder(thisx, &this->collider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, colliderBase);
+}
+
+void EnXc_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnXc* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnXc_CalculateHeadTurn(EnXc* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    this->npcInfo.unk_18 = player->actor.world.pos;
+    this->npcInfo.unk_14 = kREG(16) - 3.0f;
+    func_80034A14(&this->actor, &this->npcInfo, kREG(17) + 0xC, 2);
+}
+
+void EnXc_SetEyePattern(EnXc* this) {
+    s32 pad[3];
+    s16* blinkTimer = &this->blinkTimer;
+    s16* eyePattern = &this->eyeIdx;
+
+    if (!DECR(*blinkTimer)) {
+        *blinkTimer = Rand_S16Offset(60, 60);
+    }
+
+    *eyePattern = *blinkTimer;
+    if (*eyePattern >= ARRAY_COUNT(sEyeTextures)) {
+        *eyePattern = 0;
+    }
+}
+
+void EnXc_SpawnNut(EnXc* this, GlobalContext* globalCtx) {
+    s32 pad;
+    Vec3f* pos = &this->actor.world.pos;
+    s16 angle = this->actor.shape.rot.y;
+    f32 x = (Math_SinS(angle) * 30.0f) + pos->x;
+    f32 y = pos->y + 3.0f;
+    f32 z = (Math_CosS(angle) * 30.0f) + pos->z;
+
+    Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_ARROW, x, y, z, 0xFA0, this->actor.shape.rot.y, 0,
+                ARROW_CS_NUT);
+}
+
+void EnXc_BgCheck(EnXc* this, GlobalContext* globalCtx) {
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 75.0f, 30.0f, 30.0f, 4);
+}
+
+s32 EnXc_AnimIsFinished(EnXc* this) {
+    return SkelAnime_Update(&this->skelAnime);
+}
+
+CsCmdActorAction* EnXc_GetCsCmd(GlobalContext* globalCtx, s32 npcActionIdx) {
+    CsCmdActorAction* action = NULL;
+
+    if (globalCtx->csCtx.state != 0) {
+        action = globalCtx->csCtx.npcActions[npcActionIdx];
+    }
+    return action;
+}
+
+s32 EnXc_CompareCsAction(EnXc* this, GlobalContext* globalCtx, u16 action, s32 npcActionIdx) {
+    CsCmdActorAction* csCmdActorAction = EnXc_GetCsCmd(globalCtx, npcActionIdx);
+
+    if (csCmdActorAction && csCmdActorAction->action == action) {
```

Comment:
Relevant in a few other places
```suggestion
    if (csCmdActorAction != NULL && csCmdActorAction->action == action) {
```